MdePkg/Library/StackCheckLib/Readme.md

```diff
@@ -105,11 +105,61 @@ edk2 updated the tools_def to add `/GS` to VS2022 and VS2019 IA32/X64 builds and
 throughout the code. Every module should have a `StackCheckLib` instances linked to satisfy
 these references. So every module doesn't need to add `StackCheckLib` to the LibraryClasses
 section of the INF file, `StackCheckLib` instances should be linked as NULL in the platform
-DSC fies. The only exception to this is host-based unit tests as they will be compiled with
-the runtime libraries which already contain the stack cookie definitions and will collide
-with `StackCheckLib`.
+DSC fies. The only exception to this is MSVC builthost-based unit tests as they will be
+compiled with the runtime libraries which already contain the stack cookie definitions
+and will collide with `StackCheckLib`. A `StackCheckLibHostApplication.inf` is linked
+by `UnitTestFrameworkPkg/UnitTestFrameworkPkgHost.dsc.inc` that provides the stack
+cookie functions for GCC HOST_APPLICATIONS but not for MSVC HOST_APPLICATIONS.
+
+`MdePkg/MdeLibs.dsc.inc` links `StackCheckLibNull` for all types except SEC, HOST_APPLICATION,
+and USER_DEFINED in order to not break existing DSCs. SEC cannot be generically linked to
+because there are some SEC modules which do not link against the standard entry point
+libraries and thus do not get stack cookies inserted by the compiler. USER_DEFINED modules
+are by nature different from other modules, so we do not make any assumptions about their
+state. If a platform would like to disable stack cookies (say for debugging purposes),

```

Comment:
The code change looks good.  For the documentation I recommend adding a subsections for the following usages and put the appropriate description under each.  This way, it is more clear how a platform developer can select between the three options.

### Default Stack Check Library Configuration

### Custom Stack Check Library Configuration

### Disable Stack Check Library